src/Network/Transport/TCP/Internal.hs

```diff
@@ -52,6 +68,65 @@ import qualified Data.ByteString as BS (length, concat, null)
 import Data.Int (Int32)
 import Data.ByteString.Lazy.Internal (smallChunkSize)
 
+-- | Control headers
+data ControlHeader =
+    -- | Tell the remote endpoint that we created a new connection
+    CreatedNewConnection
+    -- | Tell the remote endpoint we will no longer be using a connection
+  | CloseConnection
+    -- | Request to close the connection (see module description)
+  | CloseSocket
+    -- | Sent by an endpoint when it is closed.
+  | CloseEndPoint
+    -- | Message sent to probe a socket
+  | ProbeSocket
+    -- | Acknowledgement of the ProbeSocket message
+  | ProbeSocketAck
+  deriving (Show)
+
+decodeControlHeader :: Word32 -> Maybe ControlHeader
+decodeControlHeader w32 = case w32 of
+  0 -> Just CreatedNewConnection
+  1 -> Just CloseConnection
+  2 -> Just CloseSocket
+  3 -> Just CloseEndPoint
+  4 -> Just ProbeSocket
+  5 -> Just ProbeSocketAck
+  _ -> Nothing
+
+encodeControlHeader :: ControlHeader -> Word32
+encodeControlHeader ch = case ch of
+  CreatedNewConnection -> 0
+  CloseConnection -> 1
+  CloseSocket -> 2
+  CloseEndPoint -> 3
+  ProbeSocket -> 4
+  ProbeSocketAck -> 5
```

Comment:
Just a trivial coding style remark: I prefer to have these sort of things aligned. Partly it's so it's easy to see the lack of silly mistakes in numeric mismatches.